src/networkapi.jl

```diff
@@ -1481,5 +1481,158 @@ function validate(rs::ReactionSystem, info::String = "")
     validated
 end
 
+"""
+    iscomplexbalanced(rs::ReactionSystem, rates::Vector)
+
+Constructively compute whether a network will have complex-balanced equilibrium
+solutions, following the method in [this paper](https://link.springer.com/article/10.1007/s10910-015-0498-2#Sec3). Accepts a map of rates [k1 => 1.0, k2 => 2.0,...]k2
+"""
+
+using Combinatorics, LinearAlgebra
+
+function iscomplexbalanced(rs::ReactionSystem, rates::Dict{Any, Float64}) 
+    if length(rates) != numparams(rs) 
+        error("The number of reaction rates must be equal to the number of parameters")
+    end
```

Comment:
```suggestion
```
See below -- this isn't how we should do the check I think since even for mass action reactions rates may not be just a single symbolic.

Reply:
I've changed it to the following - does this solve this concern? 

```julia
if length(parametermap) != numparams(rs) error... end
```

Reply:
Sure, that is fine to check.